cypress/support/api/users.js

```diff
@@ -141,6 +145,54 @@ Cypress.Commands.add('createTempUser', (permissions = [], patronGroupName, userT
               } else cy.updateRolesForUserApi(userProperties.userId, [roleId]);
             });
           } else {
+            cy.getCapabilitiesApi().then((capabilitiesResponse) => {
+              cy.getCapabilitySetsApi().then((capabilitySetsResponse) => {
+                const capabilitiesIds = [];
+                const capabilitySetsIds = [];
+                if (capabilitiesCollection.length === 0) capabilitiesCollection = [...capabilitiesResponse];
+                if (capabilitySetsCollection.length === 0) capabilitySetsCollection = [...capabilitySetsResponse];
+                permissions.forEach((permission) => {
+                  let currentPermission = '';
+                  for (const permissionObject in permissionsList) {
+                    // eslint-disable-next-line no-prototype-builtins
+                    if (permissionsList.hasOwnProperty(permissionObject)) {
+                      const { gui, internal } = permissionsList[permissionObject];
+                      if (gui.includes(permission)) {
+                        currentPermission = internal;
+                        break;
+                      }
+                    }
+                  }
+                  for (const capablityObject of capabilitiesCollection) {
+                    if (capablityObject.permission === currentPermission) {
+                      capabilitiesIds.push(capablityObject.id);
+                      break;
+                    }
+                  }
+
+                  for (const capablitySetObject of capabilitiesCollection) {
+                    if (capablitySetObject.permission === currentPermission) {
+                      capabilitySetsIds.push(capablitySetObject.id);
+                      break;
+                    }
+                  }
+                  currentPermission = '';
+                });
+
+                if (capabilitiesIds.length === 0) {
+                  cy.log('Capabilities not found ');
+                } else {
+                  cy.addCapabilitiesToNewUserApi(userProperties.userId, capabilitiesIds);
+                }
+
+                if (capabilitySetsIds.length === 0) {
+                  cy.log('Capability sets not found ');
+                } else {
+                  cy.addCapabilitySetsToNewUserApi(userProperties.userId, capabilitySetsIds);
+                }
+              });
+            });
+            cy.wait(3000);
             cy.addPermissionsToNewUserApi({
```

Comment:
For the same reason as in first comment, it would be better to leave the original method logic with permissions in `else` blick.
Also, it looks like `cy.addPermissionsToNewUserApi()` will run even on Eureka as it is now, which is not necessary